### DISCUSSION TITLE
WalletConnectConnector V2 Error

### DISCUSSION THREAD

**Discussion by gtsonevv** [wevm/wagmi#1798] [Q&A]
I get this error 
```
error - unhandledRejection: TypeError: WalletConnectProvider.init is not a function
    at WalletConnectConnector.initUniversalProvider_fn (file:///Users/georgitsonev/Desktop/building-blocks/building-blocks-ui/node_modules/@wagmi/connectors/dist/walletConnect.js:291:63)
```
when I use WalletConnectConnector with version 2. Any idea what might be causing it?
```
new WalletConnectConnector({
      chains,
      options: {
        qrcode: true,
        version: "2",
        projectId: process.env.NEXT_PUBLIC_WALLET_CONNECT_ID,
      },
}
```

**Comment by gtsonevv** [comment]
Ok, I managed to fix this one but now I cannot connect to any wallet with WalletConnect v2. Should I make some additional configuration?

**Reply by KevinLu** [reply]
How did you fix your initial error? Running into the same problem here.

**Reply by gtsonevv** [reply]
Updated wagmi to latest version.

**Comment by Mulander-J** [comment]
wagmi/core+vite+vue
same WalletConnectConnector config.
Failed scan to connect via any mobile wallets（like Trust)

**Reply by KevinLu** [reply]
Also having this error, anyone figured it out? When I paste in the wc uri generated by the web3modal into WalletConnect's OWN wallet example, it just crashes the site: https://react-wallet.walletconnect.com/

**Reply by Mulander-J** [reply]
New Update!

I change the chain parameter to mainnets which TrustWallet support.

Probably the bug is happen at Review eip155 permissions.

Former Case
```
  const { chains, provider, webSocketProvider } = configureChains(
    [
      //  Ethereum
      mainnet, goerli,
      // Polyon
      polygon, polygonMumbai,
      // Custom Chain
      chain1, chain1Dev
    ],
    [publicProvider()]
  )


    new WalletConnectConnector({
      chains,
      options: {
        qrcode: true,
        version: '2',
        projectId: import.meta.env.VITE_WC_PROJECTID
      }
    }),

```
Success Case
```
    new WalletConnectConnector({
      chains: [mainnet, polygon],
      options: {
        qrcode: true,
        version: '2',
        projectId: import.meta.env.VITE_WC_PROJECTID
      }
    }),
```

**Reply by KevinLu** [reply]
Thanks for confirming this bug, I was able to reproduce on Trust wallet as well.
Really unfortunate that some wallets don't support certain chains. This adds a lot of complexity to any dApp needing to support large number of chains + WalletConnect v2.

**Reply by KevinLu** [reply]
I was also able to reproduce the same QR code unscannable bug by adding `name` and `metadata` to the `options` object:

```js
new WalletConnectConnector({
      chains: [mainnet, polygon],
      options: {
        qrcode: true,
        version: '2',
        projectId: import.meta.env.VITE_WC_PROJECTID,
        name: 'My app',
        metadata: {
                name: 'My app',
                description: 'Test description'
        }
      }
    }),
```
